backend/src/main/java/gov/cdc/usds/simplereport/api/pxp/PatientExperienceController.java

```diff
@@ -92,7 +94,32 @@ public PxpVerifyResponse getPatientLinkVerify(
     return new PxpVerifyResponse(p, os, te);
   }
 
+  /**
+   * Given a patient link ID, returns the name of the patient for whom the test was conducted, in
+   * lightly-obfuscated form (e.g. "John Doe" -> "John D").
+   *
+   * <p>This endpoint is unauthorized and therefore does not return fully identifying data.
+   *
+   * @param request
+   * @return an obfuscated patient name
+   */
+  @GetMapping("/patient-name")
+  public String getObfuscatedPatientNameFromLink(HttpServletRequest request) {
+    String patientLink = request.getParameter("patientLink");
```

Comment:
can you use the spring way of getting a request paramerter `@RequestParam`

Reply:
You learn something new every day. Updated in 2bcf681.